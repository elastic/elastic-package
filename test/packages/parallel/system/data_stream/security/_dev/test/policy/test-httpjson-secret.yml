input: httpjson
vars:
  url: http://localhost:1234/api/v1/logs
  username: test
  password: test
  preserve_original_event: true
  enable_request_tracer: true
