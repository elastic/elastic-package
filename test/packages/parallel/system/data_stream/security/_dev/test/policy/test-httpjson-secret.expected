inputs:
    - data_stream:
        namespace: ep
      meta:
        package:
            name: system
      name: test-httpjson-secret-system
      streams:
        - auth.basic.password: ${SECRET_0}
          auth.basic.user: test
          config_version: "2"
          cursor:
            index_earliest:
                value: '[[.last_event.result.max_indextime]]'
          data_stream:
            dataset: system.security
            type: logs
          interval: 10s
          processors:
            - decode_json_fields:
                add_error_key: true
                fields: message
                target: json
            - drop_event:
                when:
                    not:
                        has_fields:
                            - json.result
            - fingerprint:
                fields:
                    - json.result._cd
                    - json.result._indextime
                    - json.result._raw
                    - json.result._time
                    - json.result.host
                    - json.result.source
                target_field: '@metadata._id'
            - drop_fields:
                fields: message
            - rename:
                fail_on_error: false
                fields:
                    - from: json.result._raw
                      to: event.original
                    - from: json.result.host
                      to: host.name
                    - from: json.result.source
                      to: event.provider
                ignore_missing: true
            - drop_fields:
                fields: json
            - decode_xml_wineventlog:
                field: event.original
                ignore_failure: true
                ignore_missing: true
                map_ecs_fields: true
                target_field: winlog
          publisher_pipeline.disable_host: true
          request.method: POST
          request.ssl: null
          request.tracer.filename: ../../logs/httpjson/http-request-trace-*.ndjson
          request.tracer.maxbackups: 5
          request.transforms:
            - set:
                target: url.params.search
                value: search sourcetype="XmlWinEventLog:Security" | streamstats max(_indextime) AS max_indextime
            - set:
                target: url.params.output_mode
                value: json
            - set:
                default: '[[(now (parseDuration "-10s")).Unix]]'
                target: url.params.index_earliest
                value: '[[ .cursor.index_earliest ]]'
            - set:
                target: url.params.index_latest
                value: '[[(now).Unix]]'
            - set:
                target: header.Content-Type
                value: application/x-www-form-urlencoded
          request.url: http://localhost:1234/api/v1/logs/services/search/jobs/export
          response.decode_as: application/x-ndjson
          tags:
            - forwarded
            - preserve_original_event
      type: httpjson
      use_output: default
output_permissions:
    default:
        _elastic_agent_checks:
            cluster:
                - monitor
        _elastic_agent_monitoring:
            indices: []
        uuid-for-permissions-on-related-indices:
            indices:
                - names:
                    - logs-system.security-ep
                  privileges:
                    - auto_configure
                    - create_doc
secret_references:
    - {}
